GCC 8 reports a warning when compiling pe_upx_stub.cpp ("strncpy() specified bound 8 equals destination size")

When building RetDec with GCC 8 on Arch Linux, it reports the following warning when compiling `pe_upx_stub.cpp`:
```
retdec/src/unpackertool/plugins/upx/pe/pe_upx_stub.cpp: In member function ‘void retdec::unpackertool::upx::PeUpxStub<bits>::fixSectionHeaders(const retdec::unpacker::DynamicBuffer&) [with int bits = 32]’:
retdec/src/unpackertool/plugins/upx/pe/pe_upx_stub.cpp:1034:10: warning: ‘char* strncpy(char*, const char*, size_t)’ specified bound 8 equals destination size [-Wstringop-truncation]
   strncpy(reinterpret_cast<char*>(newSectionHeader.Name), sectionName.c_str(), PeLib::PELIB_IMAGE_SIZEOF_SHORT_NAME);
   ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

## Steps to reproduce

```
git clone https://github.com/avast-tl/retdec
cd retdec
mkdir build
cd build
cmake ..
make retdec-unpackertool
```

## Configuration

* Commit: 14a8f4f399d04f10e5e85698b5a205b00088011d (current `master`)
* 64b Arch Linux, GCC 8.1.0, Release build of RetDec

--- a/src/unpackertool/plugins/upx/pe/pe_upx_stub.cpp
+++ b/src/unpackertool/plugins/upx/pe/pe_upx_stub.cpp
@@ -4,6 +4,9 @@
  * @copyright (c) 2017 Avast Software, licensed under the MIT license
  */
 
+#include <algorithm>
+#include <cstring>
+
 #include <pelib/PeLib.h>
 
 #include "retdec/utils/alignment.h"
@@ -1031,7 +1034,14 @@ template <int bits> void PeUpxStub<bits>::fixSectionHeaders(const DynamicBuffer&
 
 		// Parse the information about the section
 		std::string sectionName = originalHeader.readString(readPos, PeLib::PELIB_IMAGE_SIZEOF_SHORT_NAME);
-		strncpy(reinterpret_cast<char*>(newSectionHeader.Name), sectionName.c_str(), PeLib::PELIB_IMAGE_SIZEOF_SHORT_NAME);
+		// PE section names should have a fixed size of PELIB_IMAGE_SIZEOF_SHORT_NAME bytes.
+		// However, for safety reasons, ensure that there are that many bytes. If not,
+		// copy just what is available so we do not overflow the buffer.
+		const auto sectionNameSize = std::min(
+			static_cast<std::string::size_type>(PeLib::PELIB_IMAGE_SIZEOF_SHORT_NAME),
+			sectionName.size()
+		);
+		std::memcpy(newSectionHeader.Name, sectionName.data(), sectionNameSize);
 
 		// Other information about the sections are not needed so we will just skip them
 		newSectionHeader.VirtualSize        = originalHeader.read<std::uint32_t>(readPos + 8);

